From 0b731bfd29932af736938a92252aa7e594039857 Mon Sep 17 00:00:00 2001
From: Ola Jeppsson <ola@adapteva.com>
Date: Sat, 29 Aug 2015 07:53:40 +0200
Subject: [PATCH 24/33] epiphany: Emit type and size for functions

	* config/epiphany/epiphany.c (epiphany_start_function):
	Emit function type directive. Emit type and size directive for
	forwarder functions.

Signed-off-by: Ola Jeppsson <ola@adapteva.com>
---
 gcc/ChangeLog.epiphany         | 6 ++++++
 gcc/config/epiphany/epiphany.c | 3 +++
 2 files changed, 9 insertions(+)

diff --git a/gcc/ChangeLog.epiphany b/gcc/ChangeLog.epiphany
index 03ce4ce7375..cf7f15762e6 100644
--- a/gcc/ChangeLog.epiphany
+++ b/gcc/ChangeLog.epiphany
@@ -1,3 +1,9 @@
+2015-08-29  Ola Jeppsson  <ola@adapteva.com>
+
+	* config/epiphany/epiphany.c (epiphany_start_function):
+	Emit function type directive. Emit type and size directive for
+	forwarder functions.
+
 2015-08-28  Ola Jeppsson  <ola@adapteva.com>
 
 	* gcc/config/epiphany/epiphany.h: Enable -mno-soft-cmpsf under
diff --git a/gcc/config/epiphany/epiphany.c b/gcc/config/epiphany/epiphany.c
index 73e2a4ae8c6..8145b6775f7 100644
--- a/gcc/config/epiphany/epiphany.c
+++ b/gcc/config/epiphany/epiphany.c
@@ -3137,6 +3137,7 @@ epiphany_start_function (FILE *file, const char *name, tree decl)
       forwarder_attr = TREE_VALUE (TREE_VALUE (forwarder_attr));
       switch_to_section (get_section (TREE_STRING_POINTER (forwarder_attr),
 			 SECTION_CODE, decl));
+      ASM_OUTPUT_TYPE_DIRECTIVE (file, name, "function");
       ASM_OUTPUT_FUNCTION_LABEL (file, name, decl);
       if (epiphany_is_long_call_p (XEXP (DECL_RTL (decl), 0)))
 	{
@@ -3161,9 +3162,11 @@ epiphany_start_function (FILE *file, const char *name, tree decl)
 	  assemble_name (file, dst_name);
 	  fputc ('\n', file);
 	}
+      ASM_OUTPUT_MEASURED_SIZE (file, name);
       name = dst_name;
     }
   switch_to_section (function_section (decl));
+  ASM_OUTPUT_TYPE_DIRECTIVE (file, name, "function");
   ASM_OUTPUT_FUNCTION_LABEL (file, name, decl);
 }
 
-- 
2.31.1

